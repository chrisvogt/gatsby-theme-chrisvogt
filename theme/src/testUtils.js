import React from 'react'
import { Provider as ReduxProvider } from 'react-redux'
import { ThemeProvider } from 'theme-ui'
import theme from './gatsby-plugin-theme-ui'

import store from './store'

export { store }

// React module that provides necessary context to components being tested.
export const TestProvider = ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>

export const TestProviderWithState = ({ children }) => (
  <ReduxProvider store={store}>
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </ReduxProvider>
)
