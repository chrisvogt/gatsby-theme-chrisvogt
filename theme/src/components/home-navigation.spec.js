import React from 'react'
import renderer from 'react-test-renderer'

import { TestProvider } from '../testUtils'
import HomeNavigation from './home-navigation'
import useNavigationData from '../hooks/use-navigation-data'

jest.mock('../hooks/use-navigation-data')

const mockNavigationData = {
  header: {
    home: [
      {
        path: '/about',
        slug: 'about',
        text: 'About'
      },
      {
        path: '/blog',
        slug: 'blog',
        text: 'Blog'
      }
    ]
  }
}

describe('HomeNavigation', () => {
  beforeEach(() => {
    useNavigationData.mockImplementation(() => mockNavigationData)
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <TestProvider>
          <HomeNavigation />
        </TestProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('handles navigation data with empty home items', () => {
    useNavigationData.mockImplementation(() => ({
      header: {
        home: []
      }
    }))

    const tree = renderer
      .create(
        <TestProvider>
          <HomeNavigation />
        </TestProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('handles navigation data with missing header', () => {
    useNavigationData.mockImplementation(() => ({}))

    const tree = renderer
      .create(
        <TestProvider>
          <HomeNavigation />
        </TestProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('handles navigation data with null values', () => {
    useNavigationData.mockImplementation(() => null)

    const tree = renderer
      .create(
        <TestProvider>
          <HomeNavigation />
        </TestProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('handles navigation items with missing icon reactIcon', () => {
    const navigationWithMissingIcons = {
      header: {
        home: [
          {
            path: '/test',
            slug: 'test',
            text: 'Test'
          }
        ]
      }
    }
    useNavigationData.mockImplementation(() => navigationWithMissingIcons)

    const tree = renderer
      .create(
        <TestProvider>
          <HomeNavigation />
        </TestProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('handles navigation items with discogs slug for icon mapping', () => {
    const navigationWithDiscogs = {
      header: {
        home: [
          {
            path: '/discogs',
            slug: 'discogs',
            text: 'Discogs'
          }
        ]
      }
    }
    useNavigationData.mockImplementation(() => navigationWithDiscogs)

    const tree = renderer
      .create(
        <TestProvider>
          <HomeNavigation />
        </TestProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('handles navigation items with complex slug names', () => {
    const navigationWithComplexSlugs = {
      header: {
        home: [
          {
            path: '/complex-slug',
            slug: 'complex-slug',
            text: 'Complex Slug'
          }
        ]
      }
    }
    useNavigationData.mockImplementation(() => navigationWithComplexSlugs)

    const tree = renderer
      .create(
        <TestProvider>
          <HomeNavigation />
        </TestProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
