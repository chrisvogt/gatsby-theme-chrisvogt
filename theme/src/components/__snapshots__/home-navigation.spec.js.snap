// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`HomeNavigation handles navigation data with empty home items 1`] = `
<div
  className="css-h0d98v"
>
  <nav
    aria-label="On-page navigation"
    role="navigation"
  >
    <a
      className="active css-14xb3ys"
      href="#top"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-house css-1ni16ne"
        data-icon="house"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={
          {
            "height": "18px",
          }
        }
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M575.8 255.5c0 18-15 32.1-32 32.1l-32 0 .7 160.2c0 2.7-.2 5.4-.5 8.1l0 16.2c0 22.1-17.9 40-40 40l-16 0c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1L416 512l-24 0c-22.1 0-40-17.9-40-40l0-24 0-64c0-17.7-14.3-32-32-32l-64 0c-17.7 0-32 14.3-32 32l0 64 0 24c0 22.1-17.9 40-40 40l-24 0-31.9 0c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2l-16 0c-22.1 0-40-17.9-40-40l0-112c0-.9 0-1.9 .1-2.8l0-69.7-32 0c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      Home
    </a>
    <a
      className="css-14xb3ys"
      href="#posts"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-newspaper css-1ni16ne"
        data-icon="newspaper"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={
          {
            "height": "18px",
          }
        }
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 96c0-35.3 28.7-64 64-64l288 0c35.3 0 64 28.7 64 64l0 320c0 35.3-28.7 64-64 64L80 480c-44.2 0-80-35.8-80-80L0 128c0-17.7 14.3-32 32-32s32 14.3 32 32l0 272c0 8.8 7.2 16 16 16s16-7.2 16-16L96 96zm64 24l0 80c0 13.3 10.7 24 24 24l112 0c13.3 0 24-10.7 24-24l0-80c0-13.3-10.7-24-24-24L184 96c-13.3 0-24 10.7-24 24zm208-8c0 8.8 7.2 16 16 16l48 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-48 0c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16l48 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-48 0c-8.8 0-16 7.2-16 16zM160 304c0 8.8 7.2 16 16 16l256 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-256 0c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16l256 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-256 0c-8.8 0-16 7.2-16 16z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      Latest Posts
    </a>
  </nav>
</div>
`;

exports[`HomeNavigation handles navigation data with missing header 1`] = `
<div
  className="css-h0d98v"
>
  <nav
    aria-label="On-page navigation"
    role="navigation"
  >
    <a
      className="active css-14xb3ys"
      href="#top"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-house css-1ni16ne"
        data-icon="house"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={
          {
            "height": "18px",
          }
        }
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M575.8 255.5c0 18-15 32.1-32 32.1l-32 0 .7 160.2c0 2.7-.2 5.4-.5 8.1l0 16.2c0 22.1-17.9 40-40 40l-16 0c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1L416 512l-24 0c-22.1 0-40-17.9-40-40l0-24 0-64c0-17.7-14.3-32-32-32l-64 0c-17.7 0-32 14.3-32 32l0 64 0 24c0 22.1-17.9 40-40 40l-24 0-31.9 0c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2l-16 0c-22.1 0-40-17.9-40-40l0-112c0-.9 0-1.9 .1-2.8l0-69.7-32 0c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      Home
    </a>
    <a
      className="css-14xb3ys"
      href="#posts"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-newspaper css-1ni16ne"
        data-icon="newspaper"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={
          {
            "height": "18px",
          }
        }
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 96c0-35.3 28.7-64 64-64l288 0c35.3 0 64 28.7 64 64l0 320c0 35.3-28.7 64-64 64L80 480c-44.2 0-80-35.8-80-80L0 128c0-17.7 14.3-32 32-32s32 14.3 32 32l0 272c0 8.8 7.2 16 16 16s16-7.2 16-16L96 96zm64 24l0 80c0 13.3 10.7 24 24 24l112 0c13.3 0 24-10.7 24-24l0-80c0-13.3-10.7-24-24-24L184 96c-13.3 0-24 10.7-24 24zm208-8c0 8.8 7.2 16 16 16l48 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-48 0c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16l48 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-48 0c-8.8 0-16 7.2-16 16zM160 304c0 8.8 7.2 16 16 16l256 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-256 0c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16l256 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-256 0c-8.8 0-16 7.2-16 16z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      Latest Posts
    </a>
  </nav>
</div>
`;

exports[`HomeNavigation handles navigation data with null values 1`] = `
<div
  className="css-h0d98v"
>
  <nav
    aria-label="On-page navigation"
    role="navigation"
  >
    <a
      className="active css-14xb3ys"
      href="#top"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-house css-1ni16ne"
        data-icon="house"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={
          {
            "height": "18px",
          }
        }
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M575.8 255.5c0 18-15 32.1-32 32.1l-32 0 .7 160.2c0 2.7-.2 5.4-.5 8.1l0 16.2c0 22.1-17.9 40-40 40l-16 0c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1L416 512l-24 0c-22.1 0-40-17.9-40-40l0-24 0-64c0-17.7-14.3-32-32-32l-64 0c-17.7 0-32 14.3-32 32l0 64 0 24c0 22.1-17.9 40-40 40l-24 0-31.9 0c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2l-16 0c-22.1 0-40-17.9-40-40l0-112c0-.9 0-1.9 .1-2.8l0-69.7-32 0c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      Home
    </a>
    <a
      className="css-14xb3ys"
      href="#posts"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-newspaper css-1ni16ne"
        data-icon="newspaper"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={
          {
            "height": "18px",
          }
        }
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 96c0-35.3 28.7-64 64-64l288 0c35.3 0 64 28.7 64 64l0 320c0 35.3-28.7 64-64 64L80 480c-44.2 0-80-35.8-80-80L0 128c0-17.7 14.3-32 32-32s32 14.3 32 32l0 272c0 8.8 7.2 16 16 16s16-7.2 16-16L96 96zm64 24l0 80c0 13.3 10.7 24 24 24l112 0c13.3 0 24-10.7 24-24l0-80c0-13.3-10.7-24-24-24L184 96c-13.3 0-24 10.7-24 24zm208-8c0 8.8 7.2 16 16 16l48 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-48 0c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16l48 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-48 0c-8.8 0-16 7.2-16 16zM160 304c0 8.8 7.2 16 16 16l256 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-256 0c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16l256 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-256 0c-8.8 0-16 7.2-16 16z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      Latest Posts
    </a>
  </nav>
</div>
`;

exports[`HomeNavigation handles navigation items with complex slug names 1`] = `
<div
  className="css-h0d98v"
>
  <nav
    aria-label="On-page navigation"
    role="navigation"
  >
    <a
      className="active css-14xb3ys"
      href="#top"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-house css-1ni16ne"
        data-icon="house"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={
          {
            "height": "18px",
          }
        }
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M575.8 255.5c0 18-15 32.1-32 32.1l-32 0 .7 160.2c0 2.7-.2 5.4-.5 8.1l0 16.2c0 22.1-17.9 40-40 40l-16 0c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1L416 512l-24 0c-22.1 0-40-17.9-40-40l0-24 0-64c0-17.7-14.3-32-32-32l-64 0c-17.7 0-32 14.3-32 32l0 64 0 24c0 22.1-17.9 40-40 40l-24 0-31.9 0c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2l-16 0c-22.1 0-40-17.9-40-40l0-112c0-.9 0-1.9 .1-2.8l0-69.7-32 0c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      Home
    </a>
    <a
      className="css-14xb3ys"
      href="#posts"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-newspaper css-1ni16ne"
        data-icon="newspaper"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={
          {
            "height": "18px",
          }
        }
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 96c0-35.3 28.7-64 64-64l288 0c35.3 0 64 28.7 64 64l0 320c0 35.3-28.7 64-64 64L80 480c-44.2 0-80-35.8-80-80L0 128c0-17.7 14.3-32 32-32s32 14.3 32 32l0 272c0 8.8 7.2 16 16 16s16-7.2 16-16L96 96zm64 24l0 80c0 13.3 10.7 24 24 24l112 0c13.3 0 24-10.7 24-24l0-80c0-13.3-10.7-24-24-24L184 96c-13.3 0-24 10.7-24 24zm208-8c0 8.8 7.2 16 16 16l48 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-48 0c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16l48 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-48 0c-8.8 0-16 7.2-16 16zM160 304c0 8.8 7.2 16 16 16l256 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-256 0c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16l256 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-256 0c-8.8 0-16 7.2-16 16z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      Latest Posts
    </a>
    <a
      className="css-14xb3ys"
      href="/complex-slug"
    >
      Complex Slug
    </a>
  </nav>
</div>
`;

exports[`HomeNavigation handles navigation items with discogs slug for icon mapping 1`] = `
<div
  className="css-h0d98v"
>
  <nav
    aria-label="On-page navigation"
    role="navigation"
  >
    <a
      className="active css-14xb3ys"
      href="#top"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-house css-1ni16ne"
        data-icon="house"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={
          {
            "height": "18px",
          }
        }
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M575.8 255.5c0 18-15 32.1-32 32.1l-32 0 .7 160.2c0 2.7-.2 5.4-.5 8.1l0 16.2c0 22.1-17.9 40-40 40l-16 0c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1L416 512l-24 0c-22.1 0-40-17.9-40-40l0-24 0-64c0-17.7-14.3-32-32-32l-64 0c-17.7 0-32 14.3-32 32l0 64 0 24c0 22.1-17.9 40-40 40l-24 0-31.9 0c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2l-16 0c-22.1 0-40-17.9-40-40l0-112c0-.9 0-1.9 .1-2.8l0-69.7-32 0c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      Home
    </a>
    <a
      className="css-14xb3ys"
      href="#posts"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-newspaper css-1ni16ne"
        data-icon="newspaper"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={
          {
            "height": "18px",
          }
        }
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 96c0-35.3 28.7-64 64-64l288 0c35.3 0 64 28.7 64 64l0 320c0 35.3-28.7 64-64 64L80 480c-44.2 0-80-35.8-80-80L0 128c0-17.7 14.3-32 32-32s32 14.3 32 32l0 272c0 8.8 7.2 16 16 16s16-7.2 16-16L96 96zm64 24l0 80c0 13.3 10.7 24 24 24l112 0c13.3 0 24-10.7 24-24l0-80c0-13.3-10.7-24-24-24L184 96c-13.3 0-24 10.7-24 24zm208-8c0 8.8 7.2 16 16 16l48 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-48 0c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16l48 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-48 0c-8.8 0-16 7.2-16 16zM160 304c0 8.8 7.2 16 16 16l256 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-256 0c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16l256 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-256 0c-8.8 0-16 7.2-16 16z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      Latest Posts
    </a>
    <a
      className="css-14xb3ys"
      href="/discogs"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-record-vinyl css-1ni16ne"
        data-icon="record-vinyl"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={
          {
            "height": "18px",
          }
        }
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256zm256-96a96 96 0 1 1 0 192 96 96 0 1 1 0-192zm0 224a128 128 0 1 0 0-256 128 128 0 1 0 0 256zm0-96a32 32 0 1 0 0-64 32 32 0 1 0 0 64z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      Discogs
    </a>
  </nav>
</div>
`;

exports[`HomeNavigation handles navigation items with missing icon reactIcon 1`] = `
<div
  className="css-h0d98v"
>
  <nav
    aria-label="On-page navigation"
    role="navigation"
  >
    <a
      className="active css-14xb3ys"
      href="#top"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-house css-1ni16ne"
        data-icon="house"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={
          {
            "height": "18px",
          }
        }
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M575.8 255.5c0 18-15 32.1-32 32.1l-32 0 .7 160.2c0 2.7-.2 5.4-.5 8.1l0 16.2c0 22.1-17.9 40-40 40l-16 0c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1L416 512l-24 0c-22.1 0-40-17.9-40-40l0-24 0-64c0-17.7-14.3-32-32-32l-64 0c-17.7 0-32 14.3-32 32l0 64 0 24c0 22.1-17.9 40-40 40l-24 0-31.9 0c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2l-16 0c-22.1 0-40-17.9-40-40l0-112c0-.9 0-1.9 .1-2.8l0-69.7-32 0c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      Home
    </a>
    <a
      className="css-14xb3ys"
      href="#posts"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-newspaper css-1ni16ne"
        data-icon="newspaper"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={
          {
            "height": "18px",
          }
        }
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 96c0-35.3 28.7-64 64-64l288 0c35.3 0 64 28.7 64 64l0 320c0 35.3-28.7 64-64 64L80 480c-44.2 0-80-35.8-80-80L0 128c0-17.7 14.3-32 32-32s32 14.3 32 32l0 272c0 8.8 7.2 16 16 16s16-7.2 16-16L96 96zm64 24l0 80c0 13.3 10.7 24 24 24l112 0c13.3 0 24-10.7 24-24l0-80c0-13.3-10.7-24-24-24L184 96c-13.3 0-24 10.7-24 24zm208-8c0 8.8 7.2 16 16 16l48 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-48 0c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16l48 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-48 0c-8.8 0-16 7.2-16 16zM160 304c0 8.8 7.2 16 16 16l256 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-256 0c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16l256 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-256 0c-8.8 0-16 7.2-16 16z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      Latest Posts
    </a>
    <a
      className="css-14xb3ys"
      href="/test"
    >
      Test
    </a>
  </nav>
</div>
`;

exports[`HomeNavigation matches the snapshot 1`] = `
<div
  className="css-h0d98v"
>
  <nav
    aria-label="On-page navigation"
    role="navigation"
  >
    <a
      className="active css-14xb3ys"
      href="#top"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-house css-1ni16ne"
        data-icon="house"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={
          {
            "height": "18px",
          }
        }
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M575.8 255.5c0 18-15 32.1-32 32.1l-32 0 .7 160.2c0 2.7-.2 5.4-.5 8.1l0 16.2c0 22.1-17.9 40-40 40l-16 0c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1L416 512l-24 0c-22.1 0-40-17.9-40-40l0-24 0-64c0-17.7-14.3-32-32-32l-64 0c-17.7 0-32 14.3-32 32l0 64 0 24c0 22.1-17.9 40-40 40l-24 0-31.9 0c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2l-16 0c-22.1 0-40-17.9-40-40l0-112c0-.9 0-1.9 .1-2.8l0-69.7-32 0c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      Home
    </a>
    <a
      className="css-14xb3ys"
      href="#posts"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-newspaper css-1ni16ne"
        data-icon="newspaper"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={
          {
            "height": "18px",
          }
        }
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 96c0-35.3 28.7-64 64-64l288 0c35.3 0 64 28.7 64 64l0 320c0 35.3-28.7 64-64 64L80 480c-44.2 0-80-35.8-80-80L0 128c0-17.7 14.3-32 32-32s32 14.3 32 32l0 272c0 8.8 7.2 16 16 16s16-7.2 16-16L96 96zm64 24l0 80c0 13.3 10.7 24 24 24l112 0c13.3 0 24-10.7 24-24l0-80c0-13.3-10.7-24-24-24L184 96c-13.3 0-24 10.7-24 24zm208-8c0 8.8 7.2 16 16 16l48 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-48 0c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16l48 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-48 0c-8.8 0-16 7.2-16 16zM160 304c0 8.8 7.2 16 16 16l256 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-256 0c-8.8 0-16 7.2-16 16zm0 96c0 8.8 7.2 16 16 16l256 0c8.8 0 16-7.2 16-16s-7.2-16-16-16l-256 0c-8.8 0-16 7.2-16 16z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      Latest Posts
    </a>
    <a
      className="css-14xb3ys"
      href="/about"
    >
      About
    </a>
    <a
      className="css-14xb3ys"
      href="/blog"
    >
      Blog
    </a>
  </nav>
</div>
`;
